flow/scripts/global_route.tcl

```diff
@@ -1,120 +1,143 @@
-  source $env(FASTROUTE_TCL)
-} else {
-  set_global_routing_layer_adjustment $env(MIN_ROUTING_LAYER)-$env(MAX_ROUTING_LAYER) 0.5
-  set_routing_layers -signal $env(MIN_ROUTING_LAYER)-$env(MAX_ROUTING_LAYER)
-  if {[info exist env(MACRO_EXTENSION)]} {
-    set_macro_extension $env(MACRO_EXTENSION)
+# This proc is here to allow us to use 'return' to return early from this
+# file which is sourced
+proc global_route_helper {} {
+  foreach file [list $::env(RESULTS_DIR)/5_1_grt.odb \
+    $::env(REPORTS_DIR)/congestion.rpt \
+    $::env(RESULTS_DIR)/route.guide] {
+    while {[file exists $file]} {
+      file delete -force $file
+    }
   }
-}
 
-# The default behavior if the user didn't specify GLOBAL_ROUTE_ARGS is to
-# produce a drc report every 5 iterations.
-#
-# If GLOBAL_ROUTE_ARGS is specified, then we do only what the
-# GLOBAL_ROUTE_ARGS specifies.
-proc do_global_route {} {
-  set all_args [concat [list -congestion_report_file $::env(REPORTS_DIR)/congestion.rpt] \
-    [expr {[info exists ::env(GLOBAL_ROUTE_ARGS)] ? $::env(GLOBAL_ROUTE_ARGS) : \
-     {-congestion_iterations 30 -congestion_report_iter_step 5 -verbose}}]]
-
-  log_cmd global_route {*}$all_args
-}
+  source $::env(SCRIPTS_DIR)/load.tcl
+  load_design 4_cts.odb 4_cts.sdc
 
-set result [catch {do_global_route} errMsg]
-if {$result != 0} {
-  write_db $::env(RESULTS_DIR)/5_1_grt-failed.odb
-  error $errMsg
-}
+  if {[info exist ::env(PRE_GLOBAL_ROUTE)]} {
+    source $::env(PRE_GLOBAL_ROUTE)
+  }
 
-set_placement_padding -global \
-    -left $::env(CELL_PAD_IN_SITES_DETAIL_PLACEMENT) \
-    -right $::env(CELL_PAD_IN_SITES_DETAIL_PLACEMENT)
+  if {[info exist ::env(FASTROUTE_TCL)]} {
+    source $::env(FASTROUTE_TCL)
+  } else {
+    set_global_routing_layer_adjustment $::env(MIN_ROUTING_LAYER)-$::env(MAX_ROUTING_LAYER) 0.5
+    set_routing_layers -signal $::env(MIN_ROUTING_LAYER)-$::env(MAX_ROUTING_LAYER)
+    if {[info exist ::env(MACRO_EXTENSION)]} {
+      set_macro_extension $::env(MACRO_EXTENSION)
+    }
+  }
 
-set_propagated_clock [all_clocks]
-estimate_parasitics -global_routing
+  # The default behavior if the user didn't specify GLOBAL_ROUTE_ARGS is to
+  # produce a drc report every 5 iterations.
+  #
+  # If GLOBAL_ROUTE_ARGS is specified, then we do only what the
+  # GLOBAL_ROUTE_ARGS specifies.
+  proc do_global_route {} {
+    set all_args [concat [list -congestion_report_file $::env(REPORTS_DIR)/congestion.rpt] \
+      [expr {[info exists ::env(GLOBAL_ROUTE_ARGS)] ? $::env(GLOBAL_ROUTE_ARGS) : \
+      {-congestion_iterations 30 -congestion_report_iter_step 5 -verbose}}]]
+
+    log_cmd global_route {*}$all_args
+  }
 
-if {[info exist env(DONT_USE_CELLS)]} {
-  set_dont_use $::env(DONT_USE_CELLS)
-}
+  set result [catch {do_global_route} errMsg]
 
-if { ![info exists ::env(SKIP_INCREMENTAL_REPAIR)] } {
-  if {[info exist ::env(DETAILED_METRICS)]} {
-    report_metrics 5 "global route pre repair design"
-  }
+  write_guides $::env(RESULTS_DIR)/route.guide
```

Comment:
Depending on the nature of the error this may not make sense.  It would be better to do it only when you don't write a -failed.odb

Reply:
How is the user-experience diminished from having this information in the case of a failed global route?

Reply:
The same reason we write -failed.odb to indicate it isn't valid data.  Putting invalid data in a same place as valid data is confusing at best.